tests/Amazon.CloudWatch.EMF.Tests/Logger/MetricsLoggerTests.cs

```diff
@@ -354,6 +372,21 @@ public void PutMetric_WithNameTooLong_ThrowsInvalidMetricException()
             Assert.Throws<InvalidMetricException>(() => _metricsLogger.PutMetric(metricName, 1, Unit.NONE));
         }
 
+        // [Fact]
+        // public void PutMetric_WithSameMetricWithDifferentResolution_ThrowsInvalidMetricException()
+        // {
+        //       _metricsLogger.PutMetric("TestMetric", 1, Unit.COUNT,StorageResolution.Strap  );
+        //     _metricsLogger.Flush();
+        //     Assert.Throws<InvalidMetricException>(() => _metricsLogger.PutMetric(metricName, 1, Unit.NONE));
+        // }
+
+        [Fact]
+        public void PutMetric_WithSameMetricHavingDifferentResolution_ThrowsInvalidMetricException()
+        {
+              _metricsLogger.PutMetric("TestMetric", 1, Unit.COUNT,StorageResolution.STANDARD);
+              Assert.Throws<InvalidMetricException>(() =>  _metricsLogger.PutMetric("TestMetric", 1, Unit.COUNT,StorageResolution.HIGH));
+        }
+
```

Comment:
Can we add a testcase to check PutMetric_WithSameMetricHavingDifferentResolutionInDifferentFlush_ThenAllow.
if

> putMetric("Count", 1);
> logger.flush();
> logger.putMetric("Count", 1.0, StorageResolution.HIGH);

then 
Allow, log event contains resolution as high, no exception thrown

Reply:
Done